src/MSBuild/XMake.cs

```diff
@@ -2908,6 +2911,47 @@ private static void ValidateExtensions(string[] projectExtensionsToIgnore)
             }
         }
 
+        /// <summary>
+        /// Checks whether an argument given as a parameter starts with valid indicator,
+        /// <br/>which means, whether switch begins with one of: "/", "-", "--"
+        /// </summary>
+        /// <param name="unquotedCommandLineArgument">Command line argument with beginning indicator (e.g. --help).
+        /// <br/>This argument has to be unquoted, otherwise the first character will always be a quote character "</param>
+        /// <returns>true if argument's beginning matches one of possible indicators
+        /// <br/>false if argument's beginning doesn't match any of correct indicator
+        /// </returns>
+        private static bool ValidateSwitchIndicatorInUnquotedArgument(string unquotedCommandLineArgument)
+        {
+            return unquotedCommandLineArgument.StartsWith("--", StringComparison.Ordinal)
+                || unquotedCommandLineArgument.StartsWith("-", StringComparison.Ordinal)
+                || unquotedCommandLineArgument.StartsWith("/", StringComparison.Ordinal);
+        }
+
+        /// <summary>
+        /// Gets the length of the switch indicator (- or / or --)
+        /// <br/>The length returned from this method is deduced from the beginning sequence of unquoted argument.
+        /// <br/>This way it will "assume" that there's no further error (e.g. //  or ---) which would also be considered as a correct indicator.
+        /// </summary>
+        /// <param name="unquotedSwitch">Unquoted argument with leading indicator and name</param>
+        /// <returns>Correct length of used indicator
+        /// <br/>0 if no leading sequence recognized as correct indicator</returns>
+        /// Internal for testing purposes
+        internal static int GetLengthOfSwitchIndicator(string unquotedSwitch)
+        {
+            if (unquotedSwitch.StartsWith("--", StringComparison.Ordinal))
+            {
+                return "--".Length;
```

Comment:
I'd probably just put 2, 1, and 0 here to avoid allocating strings unnecessarily.

Reply:
These strings are constants and so aren't allocated. This compiles to

```il
ldstr "--"
call instance int32 [System.Private.CoreLib]System.String::get_Length()
```

However, it's probably worth avoiding that call/making the JIT constant-fold it, so I would also prefer the constants.